File: ui/lib/tauri/desktop.ts
import { invoke } from './invoke';

export async function getDesktopNotificationsEnabled(): Promise<boolean> {
  return invoke<boolean>('get_desktop_notifications_enabled');
}

export async function setDesktopNotificationsEnabled(enabled: boolean): Promise<void> {
  return invoke('set_desktop_notifications_enabled', { enabled });
}

File: tauri/src/commands/desktop.rs
use crate::tray::{self, TrayNotifyMenu};
use tauri::State;

#[tauri::command]
pub fn get_desktop_notifications_enabled() -> bool {
    tray::notifications_enabled()
}

#[tauri::command]
pub fn set_desktop_notifications_enabled(enabled: bool, menu: State<'_, TrayNotifyMenu>) {
    tray::set_notifications_enabled(enabled);
    tray::sync_notify_menu_label(&menu.0);
}
